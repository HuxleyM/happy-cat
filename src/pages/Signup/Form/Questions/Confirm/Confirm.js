import React, { useContext, useEffect } from "react";
import { UserContext } from "../../../../../Context/userContext";
import Styles from "./Confirm.module.css";

function Confirm() {
  const { user, setUser } = useContext(UserContext);
  const { userDetails, userPreference } = user;
  console.log('user', user)
  console.log(userDetails)
    useEffect(()=>{
        
    })

  return (
    <div>
      <div className={Styles.header}>
        <h3>Confirm</h3>
        <p>
          Please confirm your details before we finalize your signup. Use the
          navigation button to go backwards if things look wrong.
        </p>
      </div>
      <div className={Styles.flexContainer}>
        <h4 className={Styles.space}> UserDetails </h4>
        <div>
          <p className={Styles.bold}>User name:</p>
          <p>{user.userDetails.userName}</p>
        </div>
        <div>
          <p className={Styles.bold}>Email:</p>
          <p>{userDetails.email}</p>
        </div>
        <div>
          <p className={Styles.bold}>Password:</p>
          <p>{userDetails.password}</p>
        </div>
      </div>
      <div className={Styles.flexContainer}>
        <h4 className={Styles.space}> User Preferences </h4>
        <div>
          <p className={Styles.bold}>Include Dogs:</p>
          <p>{userPreference.allowDogs}</p>
        </div>
        <div>
          <p className={Styles.bold}>No. of GIFs per day:</p>
          <p>{userPreference.gifRate}</p>
        </div>
      </div>
    </div>
  );
}

export default Confirm;
