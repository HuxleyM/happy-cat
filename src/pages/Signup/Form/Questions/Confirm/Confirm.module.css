@import "../../../../../variables.css";

.header {
  padding: 36px 0px;
  border-bottom: 3pt solid var(--colour-grey);
  margin-bottom: 24px;
}

.header > h3 {
  font-weight: 700;
  margin-bottom: 5px;
  font-size: 20px;
}

.flexContainer {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  margin-bottom: 15px;
}

.space {
  padding-bottom: 16px;
  margin-bottom: 5px;
  border-bottom: 2pt solid var(--colour-grey);
}
.flexContainer > h4,
.bold {
  font-weight: 700;
}

.bold {
  font-size: 16px;
}

.flexContainer > div {
  margin: 10px 0px;
}
