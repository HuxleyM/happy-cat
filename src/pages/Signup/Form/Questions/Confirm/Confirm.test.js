import React from "react";
import { mount } from "enzyme";
import Confirm from "./Confirm";
import { UserContext } from "../../../../../Context/userContext";

describe("Confirm component", () => {
    const context = {
        user: {
          userDetails: {
            userName: "kipo",
            password: "theW0nder!",
            email: "kipo@wondbe.com",
          },
          userPreference: { gifRate: 4, allowDogs: "sometimes" },
        },
      };
      const wrapper = mount(
        <UserContext.Provider value={context}>
          <Confirm />
        </UserContext.Provider>
      );

  it("should render a users inputed data", () => {
    const wrapperText = wrapper.text()
    expect(wrapperText.includes('kipo')).toEqual(true)
    expect(wrapperText.includes('4')).toEqual(true)
    expect(wrapperText.includes('sometimes')).toEqual(true)
    expect(wrapperText.includes('kipo@wondbe.com')).toEqual(true)
    expect(wrapperText.includes('theW0nder!')).toEqual(true)
  });

  it('should match snapshot', ()=>{
    expect(wrapper).toMatchSnapshot();
  })
});
