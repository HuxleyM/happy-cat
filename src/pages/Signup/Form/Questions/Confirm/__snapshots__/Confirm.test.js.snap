// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm component should match snapshot 1`] = `
<Confirm>
  <div>
    <div
      className="header"
    >
      <h3>
        Confirm
      </h3>
      <p>
        Please confirm your details before we finalize your signup. Use the navigation button to go backwards if things look wrong.
      </p>
    </div>
    <div
      className="flexContainer"
    >
      <h4
        className="space"
      >
         UserDetails 
      </h4>
      <div>
        <p
          className="bold"
        >
          User name:
        </p>
        <p>
          kipo
        </p>
      </div>
      <div>
        <p
          className="bold"
        >
          Email:
        </p>
        <p>
          kipo@wondbe.com
        </p>
      </div>
      <div>
        <p
          className="bold"
        >
          Password:
        </p>
        <p>
          theW0nder!
        </p>
      </div>
    </div>
    <div
      className="flexContainer"
    >
      <h4
        className="space"
      >
         User Preferences 
      </h4>
      <div>
        <p
          className="bold"
        >
          Include Dogs:
        </p>
        <p>
          sometimes
        </p>
      </div>
      <div>
        <p
          className="bold"
        >
          No. of GIFs per day:
        </p>
        <p>
          4
        </p>
      </div>
    </div>
  </div>
</Confirm>
`;
